app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      head :temporary_redirect
+    end
+  end
+
+  def new
+    @task = Task.new(name: "Task")
+  end
+
+  def create
+    task = Task.new(task_params)
+    if task.name == "" || task.description == "" || task.completion_date == ""
+      task.save = false
+    end
+
+    is_successful = task.save
+
+    if is_successful
+      redirect_to task_path(task.id)
+    else
+      head :temporary_redirect
+    end
+  end
+
+  def edit
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+
+  def update
+    edit_task_id = params[:id].to_i
+    edited_task = Task.find_by(id: edit_task_id)
+
+    if edited_task != nil
+      edited_task.update(task_params)
+      redirect_to root_path
+    else
+      redirect_to root_path
+    end
+  end
+
+  def destroy
+    task = Task.find_by(id: params[:id])
+
+    if task.nil?
+      head :not_found
+    else
+      task.destroy
+      redirect_to tasks_path
+    end
+  end
+
+  def toggle
+    @complete_task = Task.find_by(id: params[:id])
+    # is_complete = complete_task.update(status )
+    if @complete_task.update(:status => false)
```

Comment:
This if statement is checking to see if the update method works or not, it doesn't check to see what `complete_task.status` is.

Instead I would do:

```ruby

@complete_task = Task.find_by(id: params[:id])

if @complete_task
  @complete_task.status = ! @complete_task.status
  if @complete_task.save 
    redirect_to tasks_path
  else
     head :server_error
  end
else
  head :not_found
end
```